optuna/storages/_heartbeat.py

```diff
@@ -166,8 +166,12 @@ def fail_stale_trials(study: "optuna.Study") -> None:
 
     failed_trial_ids = []
     for trial_id in storage._get_stale_trial_ids(study._study_id):
-        if storage.set_trial_state_values(trial_id, state=TrialState.FAIL):
-            failed_trial_ids.append(trial_id)
+        try:
+            if storage.set_trial_state_values(trial_id, state=TrialState.FAIL):
+                failed_trial_ids.append(trial_id)
+        except RuntimeError:
+            # If another process fails the trial, the storage raises RuntimeError.
+            pass
```

Comment:
The change itself looks good to me. 
I see some codecov warning here and I wonder if it is easy to add test for them.

Reply:
Thank you. I added a test for `RuntimeError`. Please take another look.

Reply:
Thanks for the update. LGTM